### PR TITLE
Radio button selection animation

### PR DESCRIPTION
**Pull request by pelumy**
## Issues covered
 [#1490](https://github.com/planetary-social/nos/issues/1490)
Part 3 of [#1489](https://github.com/planetary-social/nos/issues/1489)

## Description
This PR adds animation to the content flag view. Choosing a flag content category slides in the new categories underneath from the left.

## How to test
1. Build the app
2. Click the side menu
3. Click the Settings on the side menu
4. Scroll down to turn on the toggle for new moderation flow
5. Go back to the Feed screen by clicking the feed tab at the bottom of the screen
6. Click on the 3 dots on one of the notes.
7. Select `Flag this content` option. 
8. Select an option from the categories displayed.
9. Please confirm that new categories on how to flag a content shows up under the first category, and it slides in from the left.

## Screenshots/Video

https://github.com/user-attachments/assets/5f6e9f18-6f96-45f1-ad5d-ea6785148e2c

### CODE REVIEW

File: Nos/Models/FlagOption.swift
Comment (joshuatbrown) [inline Nos/Models/FlagOption.swift:1]: Thanks for fixing the end-of-line whitespace!

File: Nos/Views/Moderation/ContentFlagView.swift [hunk 1]
```diff
@@ -23,11 +23,13 @@ struct ContentFlagView: View {
                         title: String(localized: .localizable.flagContentSendTitle),
                         subtitle: nil
                     )
+                    .transition(.move(edge: .leading).combined(with: .opacity))
                 }
             }
         }
         .padding()
         .background(Color.appBg)
+        .animation(.easeInOut, value: selectedFlagOptionCategory)
```
Comment (joshuatbrown) [inline Nos/Views/Moderation/ContentFlagView.swift:32]: I'm seeing a build error on this line. Are you seeing it, too?

**Review by joshuatbrown — COMMENTED** [review]
This animation makes it so much more fun!

File: Nos/Views/Moderation/ContentFlagView.swift (hunk 1 above)
Comment (pelumy) [reply Nos/Views/Moderation/ContentFlagView.swift:32]: > I'm seeing a build error on this line. Are you seeing it, too?

Oh my God! Thanks for catching this @joshuatbrown. I removed the Equatable conformity while merging the branch with the base branch. I have pushed the fix.

**Review by joshuatbrown — APPROVED** [review]
Perfect! The code is great and so is the animation!

**Pull request merged**